core/consensus/babe/primitives/src/lib.rs

```diff
@@ -29,31 +29,28 @@ pub const BABE_ENGINE_ID: ConsensusEngineId = [b'b', b'a', b'b', b'e'];
 /// Configuration data used by the BABE consensus engine.
 #[derive(Copy, Clone, Hash, PartialEq, Eq, Debug, Encode, Decode)]
 pub struct BabeConfiguration {
-	slot_duration: u64,
-	expected_block_time: u64,
-}
+	/// The slot duration in milliseconds for BABE. Currently, only
+	/// the value provided by this type at genesis will be used.
+	///
+	/// Dynamic slot duration may be supported in the future.
+	pub slot_duration: u64,
 
-impl BabeConfiguration {
-	/// Return the expected block time in milliseconds for BABE.  Currently,
+	/// The expected block time in milliseconds for BABE. Currently,
 	/// only the value provided by this type at genesis will be used.
 	///
 	/// Dynamic expected block time may be supported in the future.
-	pub fn expected_block_time(&self) -> u64 {
-		self.expected_block_time
-	}
+	pub expected_block_time: u64,
 
-	/// Return the slot duration in milliseconds for BABE.  Currently, only
-	/// the value provided by this type at genesis will be used.
+	/// The maximum permitted VRF output, or *threshold*, for BABE. Currently,
+	/// only the value provided by this type at genesis will be used.
 	///
-	/// Dynamic slot duration may be supported in the future.
-	pub fn slot_duration(&self) -> u64 {
-		self.slot_duration
-	}
+	/// Dynamic thresholds may be supported in the future.
+	pub threshold: u64,
```

Comment:
Isn't this something that would be determined by the number of validators?

Reply:
It is indeed.  Should it be equal to `u64::MAX/(threshold * num_validators)`?